### DISCUSSION TITLE
Including specialized attributes in `@class` values of `html5` transformation output

### DISCUSSION THREAD

**Discussion by chrispy-snps** [dita-ot/dita-ot#3965] [Q&A] [answered]
We specialize some custom attribute values from `@base`, and I wanted to include their values in the `html5` transformation output so I could apply CSS styling properties for them.

For example, consider a case where I have `@foo` and `@bar` specialized attributes in my DITA source:

```
<topic id="topic">
  <title>My Topic</title>
  <body>
    <p>This is some text.</p>
    <p foo="FOO">This is some text.</p>
    <p bar="BAR">This is some text.</p>
    <p foo="FOO" bar="BAR">This is some text.</p>
    <p outputclass="abc" foo="FOO" bar="BAR">This is some text.</p>
  </body>
</topic>
```

How can I pass `@foo` and `@bar` through to the HTML5 output?

**Comment by chrispy-snps** [comment, marked as answer]
If I apply the following template in an XSLT stylesheet that extends the `dita.xsl.html5` extension point:

```
<!-- add specialized @foo and @bar attributes to @class -->
<xsl:template match="*[@foo or @bar]" mode="set-output-class" priority="10">
  <xsl:param name="default"/>
  <xsl:variable name="orig-class" as="attribute(class)">
    <xsl:next-match>
      <xsl:with-param name="default" select="$default"/>  <!-- get the original @class value -->
    </xsl:next-match>
  </xsl:variable>
  <xsl:attribute name="class" select="$orig-class, @foo, @bar" separator=" "/>  <!-- add specialized attribute values -->
</xsl:template>
```

then I get the following HTML5 output:

```
<h1 class="title topictitle1">My Topic</h1>
<div class="body">
  <p class="p">This is some text.</p>
  <p class="p FOO">This is some text.</p>
  <p class="p BAR">This is some text.</p>
  <p class="p FOO BAR">This is some text.</p>
  <p class="p abc FOO BAR">This is some text.</p>
</div>
```

and now I can apply CSS formatting:

```
.FOO { background-color: lightgreen; }
.BAR { background-color: lightblue; }
```

Here is a testcase you can experiment with: [ditaot_atts_in_outputclass.zip](https://github.com/dita-ot/dita-ot/files/9077741/ditaot_atts_in_outputclass.zip)


To try it, extract it and run the following commands:

```
# install the plugins
dita install com.synopsys.myatts.zip
dita install com.synopsys.html5.zip

# run the transformation
dita --project project.xml

# examine the output
cat out/topic.html

# uninstall the plugins
dita uninstall com.synopsys.myatts
dita uninstall com.synopsys.html5
```

**Comment by raducoravu** [comment]
As an alternative if your attributes were specialized from @props you could have stated in the ditaval file that you want them to pass through:

     <val>
       <prop action="passthrough" att="foo"/>
      </val>

and they would have been passed as "data-foo" attributes in the resulting HTML.
